controllers/notification/notification_controller_test.go

```diff
@@ -378,13 +377,15 @@ func deletionCond(msg string) toolchainv1alpha1.Condition {
 }
 
 func newController(t *testing.T, deliveryService DeliveryService,
-	initObjs ...runtime.Object) (*Reconciler, *test.FakeClient) {
+	initObjs ...runtimeclient.Object) (*Reconciler, *test.FakeClient) {
 	restore := test.SetEnvVarAndRestore(t, commonconfig.WatchNamespaceEnvVar, test.HostOperatorNs)
 	t.Cleanup(restore)
 
 	s := scheme.Scheme
 	err := apis.AddToScheme(s)
 	require.NoError(t, err)
+	// Add notification to initObjs - so that can be added as subResource
+	initObjs = append(initObjs, &toolchainv1alpha1.Notification{})
```

Comment:
Could you elaborate on this? And Notification is not a sub resource, is it?

Reply:
my understanding is that all resources with status should be added to fakeclient with `withStatusSubresource`. Then the functions update and status().update bifurcate in a way that update doesn't touch status and status().update only updates status.
here I'm adding notification in the initObjs and `NewFakeClient` adds those initObjs as StatusSubresource
this is the PR that introduces that change in controller-runtime https://github.com/kubernetes-sigs/controller-runtime/pull/2259

does that help?

Reply:
So, I took a glance at it. The fake client newly checks if there is GVK registered for the given sub-resource before it tries to update it. 
https://github.com/kubernetes-sigs/controller-runtime/blob/735b6073bb253c0449bfcf6641855dcf2118bb15/pkg/client/fake/client.go#L413
But default, the client registers GVKs for hard-coded list of resources: 
https://github.com/kubernetes-sigs/controller-runtime/blob/735b6073bb253c0449bfcf6641855dcf2118bb15/pkg/client/fake/client.go#L1240-L1274
and for those that were added via method `WithStatusSubresource`
https://github.com/kubernetes-sigs/controller-runtime/blob/release-0.15/pkg/client/fake/client.go#L207-L210
which is something Kanika does in her toolchain-common PR for all objects included provided as init object here:
https://github.com/codeready-toolchain/toolchain-common/pull/376/files#diff-1e288d578a8b0f65d3d7f6895c7574365de50afcfe1fd83de1fa2c32819c99e0R26
and because the Notification CR wasn't provided as an init object, then it was registered as containing a sub-resource.

TBH, there is a room for an improvement in our implementation of the fake client. I'll leave a comment there.

Reply:
https://github.com/codeready-toolchain/toolchain-common/pull/376/files#r1671840863